tests/foreman/api/test_host.py

```diff
@@ -1387,6 +1387,23 @@ def test_positive_read_puppet_ca_proxy_name(
     assert session_puppet_enabled_proxy.name == host['puppet_ca_proxy_name']
 
 
+@pytest.mark.tier2
+def test_positive_list_hosts_thin_all(module_target_sat):
+    """List hosts with thin=true and per_page=all
+
+    :id: 00b7e603-aed5-4b19-bfec-1a179fad6743
+
+    :expectedresults: Hosts listed without ISE
+
+    :BZ: 1969263
+
+    :customerscenario: true
+
+    :CaseImportance: High
+    """
+    module_target_sat.api.Host().search(query={'thin': 'true', 'per_page': 'all'})
```

Comment:
Can we add some validations after this search to verify if a thin host record is returned correctly?

Reply:
This already checks that there is no ISE but yes, I could add some check for data validity.